app/middleware/services.js

```diff
@@ -152,34 +153,43 @@ function getDisplayValuesMapper(location) {
 }
 
 function prepareForRender(req, res, next) {
-  const tenClosestPlaces = req.pharmacyList
-        .sort(sortByDistance)
-        .slice(0, 10);
-  const serviceList = pharmacyMapper(tenClosestPlaces);
+  const open = req.query.open || false;
   const location = req.query.location;
-
-  // eslint-disable-next-line no-param-reassign
-  req.serviceList =
-    serviceList
-      .map(getDisplayValuesMapper(location))
-      .sort(sortByDistanceInKms);
-
-  next();
-}
-
-function prepareOpenThingsForRender(req, res, next) {
-  // Only get the 2 closet OPEN pharmacies
-  const serviceLimit = 2;
-  const location = req.query.location;
-  // eslint-disable-next-line no-param-reassign
-  req.serviceList =
-    pharmacyMapper(req.pharmacyList)
-      .sort(sortByDistanceInKms)
-      .filter((pharmacy) => (pharmacy.openingTimes ?
-         pharmacy.openingTimes.isOpen(moment()) :
-         false))
-      .slice(0, serviceLimit)
-      .map(getDisplayValuesMapper(location));
+  let serviceList = [];
+  let altResultsUrl = '';
+  let altResultsMessage = '';
+
+  if (open) {
+    altResultsUrl = `results?location=${location}`;
+    altResultsMessage = 'See all nearby places that can help, open or closed';
+
+    serviceList =
+      pharmacyMapper(req.pharmacyList)
+        .sort(sortByDistanceInKms)
+        .filter((pharmacy) => (pharmacy.openingTimes ?
+           pharmacy.openingTimes.isOpen(moment()) :
+           false))
+        .slice(0, 2)
+        .map(getDisplayValuesMapper(location));
+  } else {
+    altResultsUrl = `results?location=${location}&open=true`;
+    altResultsMessage = 'See only open places nearby';
+
+    const tenClosestPlaces = req.pharmacyList
+          .sort(sortByDistance)
+          .slice(0, 10);
+
+    serviceList =
+      pharmacyMapper(tenClosestPlaces)
+        .map(getDisplayValuesMapper(location))
+        .sort(sortByDistanceInKms);
```

Comment:
TBH, I'd not looked at the detail in this change, I was focussing on moving stuff around to get rid of the 2 methods. Although `prepareOpenThingsForRender` had only a single sort `prepareForRender` still had 2 which is why this branch has...

Is it still required? Not that I can see.

Do you want it removing as part of this or the next refactor?